src/main/java/org/opentripplanner/model/plan/legreference/ScheduledTransitLegReference.java

```diff
@@ -23,30 +25,70 @@ public record ScheduledTransitLegReference(
   int toStopPositionInPattern
 )
   implements LegReference {
+  private static final Logger LOG = LoggerFactory.getLogger(ScheduledTransitLegReference.class);
+
+  /**
+   * Reconstruct a scheduled transit leg from this scheduled transit leg reference.
+   * Since the transit model could have been modified between the time the reference is created
+   * and the time the transit leg is reconstructed (either because new planned data have been
+   * rolled out, or because a realtime update has modified a trip),
+   * it may not be possible to reconstruct the leg.
+   * In this case the method returns null.
+   */
   @Override
   public ScheduledTransitLeg getLeg(TransitService transitService) {
```

Comment:
Add `Nullable` annotation or return an `Optional`, please.

Reply:
Done

Reply:
It makes me sad to see that an entity has a dependency on a Service, but +1 for adding the doc. We have a rather critical bug here #5422, so no need to fix it. We can see if we can refactor, when fixing the bug. But, due to this the class become unnecessary complicated to test.

The way to resolve this is to change the method to something like this (the arguments are just exampls):

```
public ScheduledTransitLeg constructLeg(Trip trip, TripPattern pattern, TimeTable table) { ...
```
then the wiring would eiter go into a Service or the caller - depending on the need for reusability.